enumerables.rb

```diff
@@ -0,0 +1,332 @@
+# rubocop:disable Metrics/CyclomaticComplexity, Metrics/MethodLength, Metrics/PerceivedComplexity, Lint/Void
+p '-----------|||||--------'
+p '----------{(o o)}--------'
+p '--------ooO-(_)-Ooo------'
+# Initial comment
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    array = to_a
+    size.times { |index| yield(array[index]) }
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    array = to_a
+    size.times { |index| yield(array[index], index) }
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    array = to_a
+    arraya = []
+    array.length.times do |a|
+      array[a]
+      arraya << array[a] if yield (array[a])
+    end
+    arraya
+  end
+
+  def my_all?(*arg)
+    return 'error arguments' if arg.length > 1
+
+    if block_given?
+      array = to_a
+      cont = 0
+      array.length.times do |a|
+        array[a]
+        if yield (array[a])
+          cont += 1
+          return true if cont == array.length
+
+          next
+        end
+        return false
+      end
+      true elsif arg.empty?
+    elsif arg[0].class == Class
+      array3 = to_a
+      cont2 = 0
+      array3.length.times do |b|
+        if array3[b].is_a?(arg[0])
+          cont2 += 1
+          return true if cont2 == array3.length
+
+          next
+        end
+        return false
+      end
+    elsif arg[0].class == Regexp
+      array3 = to_a
+      cont2 = 0
+      array3.length.times do |b|
+        if arg[0].match(array3[b])
+          cont2 += 1
+          return true if cont2 == array3.length
+
+          next
+        end
+        return false
+      end
+    end
+  end
+end
+
+module Enumerable
+  def my_any?(*arg)
+    return 'error arguments' if arg.length > 1
+
+    if block_given?
+      array = to_a
+      array.length.times do |a|
+        array[a]
+        return true if yield (array[a])
+
+        return false
+      end
+      true elsif arg.empty?
+
+    elsif arg[0].class == Class
+      array3 = to_a
+      array3.length.times do |b|
+        return true if array3[b].is_a?(arg[0])
+
+        return false
+      end
+    elsif arg[0].class == Regexp
+      array3 = to_a
+
+      array3.length.times do |b|
+        return true if arg[0].match(array3[b])
+
+        return false
+      end
+    end
+  end
+
+  def my_none?(*arg)
```

Comment:
Should returns false if the block ever returns true for all elements::range
![Screen Shot 2020-07-21 at 7 48 10 PM](https://user-images.githubusercontent.com/52430951/88121459-5d093a80-cb8b-11ea-9715-e174c9656a85.png)
****
When no block or argument is given returns true only if none of the collection members is true.
![Screen Shot 2020-07-21 at 7 51 22 PM](https://user-images.githubusercontent.com/52430951/88121519-92ae2380-cb8b-11ea-89c4-ff106942687b.png)
****
When a pattern other than Regex or a Class is given returns false only if one of the collection matches the pattern
![Screen Shot 2020-07-21 at 7 53 13 PM](https://user-images.githubusercontent.com/52430951/88121589-d6089200-cb8b-11ea-9d11-10e643603d63.png)

Reply:
##specify argument true_block return and error also with the original method
![code](https://user-images.githubusercontent.com/53111464/88134583-a3ba5d00-cbaa-11ea-90bd-83f19a828bcc.png)

Traceback (most recent call last):
enumerables.rb:400:in `<main>': undefined local variable or method `true_block' for main:Object (NameError)